### PR TITLE
[SC-4213] Create embedding metrics

### PR DESCRIPTION
**Pull request by juanmleng**
## Internal Notes for Reviewers

- Add a new notebook `rag_vendor_contracts_llama.ipynb` that includes new embedding metrics and uses the vendor contracts dataset.
- The notebook includes two types of CSV loaders, using `llama_index` and `langchain`, with different characteristics.
- This is a WIP notebook. Remaining tasks include developing metrics for chunking and adding more metrics for embeddings.

### CODE REVIEW

**Review by johnwalz97 — APPROVED** [review]
looking good so far. 👌

File: validmind/tests/model_validation/embeddings/CosineSimilarityHeatmap.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    # Create the heatmap using Plotly
+    fig = px.imshow(
+        similarity_matrix,
+        labels=dict(x=xaxis_title, y=yaxis_title, color=color),
+        x=indices,
+        y=indices,
+        text_auto=True,
+        aspect="auto",
+        color_continuous_scale=color_scale,
+    )
```
Comment (cachafla) [inline validmind/tests/model_validation/embeddings/CosineSimilarityHeatmap.py:50]: It seems like the `Index` labels might not be too relevant to show on the plot axis (x and y)? I wonder if it improves the visualization as we compute the metric for many more rows from the dataset.

Comment (cachafla) [reply validmind/tests/model_validation/embeddings/CosineSimilarityHeatmap.py:50]: Comment was redirected to this `indices = [f"{xaxis_title} {i+1}" for i in range(len(dataset.df))]`

File: validmind/tests/model_validation/embeddings/EuclideanDistanceHeatmap.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    Returns:
+        None: Displays an interactive heatmap.
+    """
+    embeddings = np.stack(dataset.y_pred(model))
+
+    # Calculate pairwise Euclidean distance
+    distance_matrix = euclidean_distances(embeddings)
+
+    # Create labels for axes
+    indices = [f"{xaxis_title} {i+1}" for i in range(len(dataset.df))]
```
Comment (cachafla) [inline validmind/tests/model_validation/embeddings/EuclideanDistanceHeatmap.py:40]: Same question

**Review by cachafla — APPROVED** [review]
Nice 👌 I left a comment suggesting if omitting the index labels could improve readability of the plot when there are too many rows in the dataset (more than 20 or something).

**Review by cachafla — APPROVED** [review]

**Pull request merged**